pureconfig/src/main/scala/com/avast/clients/rabbitmq/pureconfig/Implicits.scala

```diff
@@ -0,0 +1,96 @@
+package com.avast.clients.rabbitmq.pureconfig
+
+import _root_.pureconfig.ConfigReader.Result
+import _root_.pureconfig._
+import _root_.pureconfig.generic.ProductHint
+import _root_.pureconfig.generic.semiauto._
+import com.avast.clients.rabbitmq.api.DeliveryResult
+import com.avast.clients.rabbitmq.api.DeliveryResult._
+import com.avast.clients.rabbitmq.{
+  AutoBindExchangeConfig,
+  AutoBindQueueConfig,
+  AutoDeclareExchangeConfig,
+  AutoDeclareQueueConfig,
+  BindArgumentsConfig,
+  BindExchangeConfig,
+  BindQueueConfig,
+  ConsumerConfig,
+  CredentialsConfig,
+  DeclareArgumentsConfig,
+  DeclareExchangeConfig,
+  DeclareQueueConfig,
+  ExchangeType,
+  NetworkRecoveryConfig,
+  ProducerConfig,
+  ProducerPropertiesConfig,
+  PullConsumerConfig,
+  RabbitMQConnectionConfig,
+  RecoveryDelayHandlers,
+  SslConfig,
+  TrustStoreConfig
+}
+import com.rabbitmq.client.RecoveryDelayHandler
+import org.slf4j.event.Level
+
+object Implicits extends Implicits
+
+trait Implicits {
+  // use camelCase names (see https://pureconfig.github.io/docs/overriding-behavior-for-case-classes.html#field-mappings)
+  private implicit def hint[T]: ProductHint[T] = ProductHint[T](ConfigFieldMapping(CamelCase, CamelCase))
```

Comment:
Shouldn't this be configurable? You are forcing everyone to use your convention.

Reply:
OK, good idea, however the default has to be this, to keep backward compatibility. Not sure how to achieve that, will think about that.

Reply:
Yes